### ISSUE TITLE
aws_appsync_alpha: Support for eventbridge datasource

### ISSUE THREAD

**Issue by ben-elsen** [aws/aws-cdk#25353]
### Describe the feature

I would like to be able to use the eventbridge datasource type with the aws_appsync_alpha module. At the moment, this seems impossible. 

### Use Case

I want to be able to publish events to eventbridge from my appsync api. This is a very common pattern in my application. 

### Proposed Solution

Extend the BaseDataSource class to handle the eventbridge datasource type and add a add_eventbridge_data_source function to the GraphqlApi class.

### Other Information

I tried to use the CfnDataSource class from aws_appsync in combo with AppsyncFunction from the aws_appsync_alpha module but that throws me the error: 

TypeError: type of argument data_source must be aws_cdk.aws_appsync_alpha.BaseDataSource; got aws_cdk.aws_appsync.CfnDataSource instead

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.77.0 (build 06a0b19)

### Environment details (OS name and version, etc.)

MacOs Ventura 13.3.1 (22E261)

**Comment by pahud** [comment]
Yeah we probably need to create a new data source class by extending BaseDataSource. Making this a p2 feature request. Any PRs or upvotes are welcome and appreciated.

**Comment by postsa** [comment]
Is this a duplicate of https://github.com/aws/aws-cdk/issues/24809?

**Comment by TheRealAmazonKendra** [comment]
AppSync has now been graduated to stable so the alpha package is no longer being updated.
